pkg/sfu/sendsidebwe/packet_tracker.go

```diff
@@ -0,0 +1,113 @@
+package sendsidebwe
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/livekit/protocol/logger"
+)
+
+// -------------------------------------------------------------------------------
+
+var (
+	errNoPacketInRange = errors.New("no packet in range")
+)
+
+// -------------------------------------------------------------------------------
+
+type packetTrackerParams struct {
+	Logger logger.Logger
+}
+
+type packetTracker struct {
+	params packetTrackerParams
+
+	lock sync.Mutex
```

Comment:
Unfortunately no. I initially relied on the wrapping lock (from `congestion_detector`), but in the forward path, when sequence numbers are created, `congestion_detector` is not in the path.

Also, in the reverse path, did not want to rely on the wrapping lock as it would mean holding the lock while the report is processed (it could be fast, but it would still affect the forward path).

Reply:
oh, i missed that it's part of the public interface of congestion detector...